### PR TITLE
Add version information to hostfxr component

### PR DESCRIPTION
**Pull request by schellap**
Add correct version info for the hostfxr component ID.

### CODE REVIEW

File: build_projects/dotnet-host-build/PkgTargets.cs [hunk 1]
```diff
@@ -31,7 +31,9 @@ public static BuildTargetResult InitPkg(BuildTargetContext c)
             Directory.CreateDirectory(PkgsIntermediateDir);
 
             SharedHostComponentId = $"com.microsoft.dotnet.sharedhost.component.osx.x64";
-            HostFxrComponentId = $"com.microsoft.dotnet.hostfxr.component.osx.x64";
+
+            string hostFxrVersion = c.BuildContext.Get<HostVersion>("HostVersion").LockedHostFxrVersion.ToString();
```
Comment (gkhanna79) [inline build_projects/dotnet-host-build/PkgTargets.cs]: Have you confirmed other packages do not have this problem and that with this fix, the issue no longer repros?

**Comment by gkhanna79** [comment]
Fixes https://github.com/dotnet/core-setup/issues/444

**Comment by mellinoe** [comment]
To clarify, this variable controls what this templated string is replaced with, correct?

https://github.com/dotnet/core-setup/blob/master/packaging/osx/sharedframework/shared-framework-distribution-template.xml#L20

Previously, this ID was unversioned, so any sharedframework pkg installer would overwrite that folder because they had the same ID. Now, they will have distinct ID's with different version substrings.

Is that an accurate description of the fix?

**Comment by gkhanna79** [comment]
Adding @brthor as well for review.

**Comment by brthor** [comment]
@mellinoe That's an accurate description as I understand it. Assuming the shared framework is installing SxS this causes hostfxr to mimic that. 

LGTM.

**Comment by schellap** [comment]
Not for review yet. Closing...

File: build_projects/dotnet-host-build/PkgTargets.cs (hunk 1 above)
Comment (schellap) [inline build_projects/dotnet-host-build/PkgTargets.cs]: Yes, it looks good.

**Comment by gkhanna79** [comment]
LGTM as well.

**Comment by mellinoe** [comment]
@schellap The commit seems to be made with the wrong author email.

**Pull request merged**